cpp/src/copying/contiguous_split.cu

```diff
@@ -89,16 +90,21 @@ struct src_buf_info {
  * M partitions, then we have N*M destination buffers.
  */
 struct dst_buf_info {
+  // constant across all copy commands for this buffer
   std::size_t buf_size;  // total size of buffer, including padding
   int num_elements;      // # of elements to be copied
   int element_size;      // size of each element in bytes
-  int num_rows;  // # of rows (which may be different from num_elements in the case of validity or
-                 // offset buffers)
-  int src_row_index;       // row index to start reading from from my associated source buffer
+  int num_rows;  // # of rows to be copied(which may be different from num_elements in the case of
```

Comment:
Maybe out of scope for this PR, but the types of a lot of these variables seem like they should be something `cudf::size_type` or at least something more semantically meaningful. Otherwise it's not clear to me why the buffer size is bound by the range of `int`.